packages/make-styles-webpack-loader/src/transformSync.ts

```diff
@@ -0,0 +1,47 @@
+import * as Babel from '@babel/core';
+import babelPluginMakeStyles from '@fluentui/babel-make-styles';
+
+export type TransformOptions = {
+  filename: string;
+
+  inputSourceMap: Babel.TransformOptions['inputSourceMap'];
+  enableSourceMaps: boolean;
+};
+
+export type TransformResult = {
+  code: string;
+  sourceMap: NonNullable<Babel.BabelFileResult['map']> | undefined;
+};
+
+/**
+ * Transforms passed source code with Babel, uses user's config for parsing, but ignores it for transforms.
+ */
+export function transformSync(sourceCode: string, options: TransformOptions): TransformResult {
+  // Parse the code first so Babel will use user's babel config for parsing
+  // During transforms we don't want to use user's config
+  const ast = Babel.parseSync(sourceCode, {
+    caller: { name: 'make-styles' },
+
+    filename: options.filename,
+    inputSourceMap: options.inputSourceMap,
+    sourceMaps: options.enableSourceMaps,
+  })!;
```

Comment:
can we stop using deprecated typescript features ? Instead we should handle the logic accordingly


```
if(ast === null){
  throw new Error(...)
}

const babelFileResult = ...

if(babelFileResult === null){
  throw new Error(...)
}
```

Reply:
> Can you please point me in Babel docs when it returns null?

I was following typings only ⌨️
![2021-07-15 at 12 23](https://user-images.githubusercontent.com/1223799/125772995-58f5617c-6432-4a43-a949-07fe150574f3.png)

Reply:
this time the 3rd party types are actually correct ! :D https://github.com/babel/babel/blob/main/packages/babel-core/src/parse.ts#L25

Reply:
Updated in 86c7e6f.